src/hooks/screen/useChatScreenHook.ts

```diff
@@ -103,7 +122,7 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       if (e?.response?.data?.status === 'Channel is blocked') return;
 
       setTimeout(() => {
-        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+        sendChat(message, attachments, iteration + 1, currentChatSchema).catch((sendChatError) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendChat` function is being called recursively with the same `attachments` parameter. If there's an error with one of the attachments, this could lead to an infinite loop of failed attempts. Consider adding a mechanism to handle or skip faulty attachments after a certain number of retries.

```diff
-         sendChat(message, attachments, iteration + 1, currentChatSchema).catch((sendChatError) => {
+         if (iteration < MAX_RETRIES) {
+           sendChat(message, attachments, iteration + 1, currentChatSchema).catch((sendChatError) => {
+             console.log(sendChatError);
+           });
+         } else {
+           console.error('Max retries reached for sending chat message');
+         }
```
Where `MAX_RETRIES` is a constant that defines the maximum number of retry attempts.


<!-- This is an auto-generated comment by OSS CodeRabbit -->